pkg/forward/metrics.go

```diff
@@ -23,123 +23,123 @@ import (
 	"github.com/numaproj/numaflow/pkg/metrics"
 )
```

Comment:
Why are we exporting the metric names? If source and forwarder metrics are going to be different, we can just make a copy and delete it later. IMHO, they should be different, easy to search by metric name rather than by tags.

Reply:
then just copy over the metrics file for the time being without exporting.

Reply:
It seems not allowed to register duplicate metrics
```
/Users/kyang5/sdk/go1.19.2/bin/go tool test2json -t /Users/kyang5/Library/Caches/JetBrains/GoLand2023.1/tmp/GoLand/___data_forward_test_go.test -test.v -test.paniconexit0 -test.run ^\QTestNewDataForward\E|\QTestNewDataForwardIdleWatermark\E|\QTestNewDataForwardIdleWatermark_Reset\E|\QTestDataForwardSinglePartition\E|\QTestDataForwardMultiplePartition\E|\QTestWriteToBuffer\E$
panic: duplicate metrics collector registration attempted

goroutine 1 [running]:
github.com/prometheus/client_golang/prometheus.(*Registry).MustRegister(0xc000195680?, {0xc0001bb0e0?, 0x1, 0x0?})
	/Users/kyang5/go/pkg/mod/github.com/prometheus/client_golang@v1.12.1/prometheus/registry.go:403 +0x7f
github.com/prometheus/client_golang/prometheus/promauto.Factory.NewCounterVec({{0x1eaabb0?, 0xc000041a90?}}, {{0x0, 0x0}, {0x1ce19e9, 0x9}, {0x1ce2efb, 0xa}, {0x1cf8656, 0x1d}, ...}, ...)
	/Users/kyang5/go/pkg/mod/github.com/prometheus/client_golang@v1.12.1/prometheus/promauto/auto.go:276 +0x133
github.com/prometheus/client_golang/prometheus/promauto.NewCounterVec(...)
	/Users/kyang5/go/pkg/mod/github.com/prometheus/client_golang@v1.12.1/prometheus/promauto/auto.go:176
github.com/numaproj/numaflow/pkg/sources/forward.init()
	/Users/kyang5/Desktop/development/numaflow/numaflow/pkg/sources/forward/metrics.go:27 +0x152
```

Reply:
Should we consider using `source_data_forwarder` instead of `forwarder` as new metric name? It's as you mentioned above easier to search but backward incompatible.

Reply:
i am okay with not being backward incompatible. i want to clean out the code before we hit 1.0, which is by the end of Sept. WDYT?

Reply:
Agree. I don't know how many customers are actively using the existing source metric to build their dashboards. It should be straight-forward to update the dashboard accordingly after upgrading numaflow to 1.0, we just need to communicate it to customers.